File: Program's_Contributed_By_Contributors/C++_Programs/Arrays/Two Sum.cpp
// Link to Problem : https://leetcode.com/problems/two-sum/

vector<int> twoSum(vector<int>& num, int target) {
      map<int,int> m;
      vector<int> ans(2,0);
      int a;
      for(int i=0;i<num.size();i++){
          a = target-num[i];
          auto it = m.find(a);
          if(it != m.end()){
              ans[0]=it->second;
              ans[1]=i;
              return ans;
          }
          else
              m[num[i]]=i;
      }
      return ans;
  }

File: Program's_Contributed_By_Contributors/Python_Programs/twosum.py
def two_sum(nums, target):
    num_map = {}
    
    for index, num in enumerate(nums):
        complement = target - num
        if complement in num_map:
            return [num_map[complement], index]
        num_map[num] = index
        
    return None

# Example usage:
nums = [2, 7, 11, 15]
target = 9
result = two_sum(nums, target)

if result:
    print(f"Indices: {result[0]}, {result[1]}")
else:
    print("No two sum solution found.")
